Custom childrens for the thumbnails

KMouratidis opened the pull request:
As discussed in https://github.com/yifaneye/react-gallery-carousel/issues/40, this should allow users to pass their own thumbnails.

yifaneye commented:
Hi @KMouratidis ! Thank you for your contribution!

Your pull request looks good to me! There are 2 things that I would like to point out:
Would you like to add an example in e.g. `example/src/components/Carousel5.js`? 
Would you like to choose `yifaneye:dev` instead of `yifaneye:master` as the destination branch? 

I am happy to merge it afterwards.

KMouratidis commented:
Rebased and added an example which seems to work (you'll still need to create the actual link to the code):

![image](https://user-images.githubusercontent.com/26832680/133035300-0545ec9f-3002-4747-80c6-40797d6df50a.png)

Let me know if there's more stuff that needs changing!

yifaneye commented:
Hi @KMouratidis ! Thanks! I will take a look tomorrow.

yifaneye commented:
Hi @KMouratidis ! I had a look just now! Your work is great! I am halfway through the review.
![SS 2021-09-16 at 01 22 13](https://user-images.githubusercontent.com/49315663/133462064-cef4103c-fcf3-4fb3-971b-26608ef78996.jpg)

yifaneye commented on src/components/Carousel/Carousel.js:
```diff
@@ -55,6 +55,7 @@ const GalleryCarousel = (props, ref) => {
     index: props.index,
     isLoop: props.isLoop
   });
+  const thumbnailElements = props.thumbnails || slidesElements;
```
`slideElements` sounds better than `slidesElements`. I will rename `slidesElements` after your changes. You do not need to change it!

yifaneye replied:
Please add `thumbnails: PropTypes.arrayOf(PropTypes.node),` to `src/components/Carousel/props.js`.

yifaneye reviewed (changes requested):
Thanks for the contribution! Would you like to address the minor comments?

yifaneye commented on example/src/components/Carousel6.js:
```diff
@@ -0,0 +1,33 @@
+import 'react-gallery-carousel/dist/index.css';
+
+const Carousel6 = ({ images }) => {
+  const thumbnails = images.map((_, idx) => (
```
It will look like this.
![SS 2021-10-04 at 00 38 29](https://user-images.githubusercontent.com/49315663/135756046-f427b790-8b9a-4003-a7f9-f4b05fd5ca93.jpg)

KMouratidis commented:
I just added your suggestions - thank you! How does it look?

yifaneye commented on example/src/components/Carousel6.js:
```diff
@@ -0,0 +1,33 @@
+  ));
+  const imageElements = images.map((image) => (
+    <img src={image.src} alt={image.alt} height={700} width={1200} />
+  ));
```
I got the below 'unique "key" prop' warning.
![SS 2021-10-06 at 00 21 15](https://user-images.githubusercontent.com/49315663/136031354-a5ef9f65-39bb-43bb-893f-f285061335dd.jpg)
Please change to:
```
const imageElements = images.map((image, index) => (
    <img
      src={image.src}
      alt={image.alt}
      className='image-responsive'
      key={index}
    />
));
```

yifaneye reviewed (commented):
Hi @KMouratidis ! It looks good to me! Would you like to address 1 additional minor comment?
